qiskit/providers/basic_provider/basic_simulator.py

```diff
@@ -35,99 +37,71 @@
 
-    DEFAULT_CONFIGURATION = {
-        "backend_name": "qasm_simulator",
-        "backend_version": "2.1.0",
-        "n_qubits": min(24, MAX_QUBITS_MEMORY),
-        "url": "https://github.com/Qiskit/qiskit-terra",
-        "simulator": True,
-        "local": True,
-        "conditional": True,
-        "open_pulse": False,
-        "memory": True,
-        "max_shots": 0,
-        "coupling_map": None,
-        "description": "A python simulator for qasm experiments",
-        "basis_gates": ["h", "u", "p", "u1", "u2", "u3", "rz", "sx", "x", "cx", "id", "unitary"],
-        "gates": [
-            {
-                "name": "h",
-                "parameters": [],
-                "qasm_def": "gate h q { U(pi/2,0,pi) q; }",
-            },
-            {
-                "name": "p",
-                "parameters": ["lambda"],
-                "qasm_def": "gate p(lambda) q { U(0,0,lambda) q; }",
-            },
-            {
-                "name": "u",
-                "parameters": ["theta", "phi", "lambda"],
-                "qasm_def": "gate u(theta,phi,lambda) q { U(theta,phi,lambda) q; }",
-            },
-            {
-                "name": "u1",
-                "parameters": ["lambda"],
-                "qasm_def": "gate u1(lambda) q { U(0,0,lambda) q; }",
-            },
-            {
-                "name": "u2",
-                "parameters": ["phi", "lambda"],
-                "qasm_def": "gate u2(phi,lambda) q { U(pi/2,phi,lambda) q; }",
-            },
-            {
-                "name": "u3",
-                "parameters": ["theta", "phi", "lambda"],
-                "qasm_def": "gate u3(theta,phi,lambda) q { U(theta,phi,lambda) q; }",
-            },
-            {"name": "rz", "parameters": ["phi"], "qasm_def": "gate rz(phi) q { U(0,0,phi) q; }"},
-            {
-                "name": "sx",
-                "parameters": [],
-                "qasm_def": "gate sx(phi) q { U(pi/2,7*pi/2,pi/2) q; }",
-            },
-            {"name": "x", "parameters": [], "qasm_def": "gate x q { U(pi,7*pi/2,pi/2) q; }"},
-            {"name": "cx", "parameters": [], "qasm_def": "gate cx c,t { CX c,t; }"},
-            {"name": "id", "parameters": [], "qasm_def": "gate id a { U(0,0,0) a; }"},
-            {"name": "unitary", "parameters": ["matrix"], "qasm_def": "unitary(matrix) q1, q2,..."},
-        ],
-    }
-
-    DEFAULT_OPTIONS = {"initial_statevector": None, "chop_threshold": 1e-15}
-
-    # Class level variable to return the final state at the end of simulation
-    # This should be set to True for the statevector simulator
-    SHOW_FINAL_STATE = False
-
-    def __init__(self, configuration=None, provider=None, **fields):
+    def __init__(
+        self,
+        provider: Provider | None = None,
+        target: Target | None = None,
+        **fields,
+    ) -> None:
+
+        """Initialize a :class:`.BackendV2`-based :class:`.BasicSimulator`.
+
```

Comment:
Oftentimes we remove the first line if it doesn't state something special. It also looks good in the docs since after the classname the args are directly listed 🙂 Just an optional suggestion!
```suggestion
        """
```